amazon/ionbenchmark/ion_benchmark_cli.py

```diff
@@ -1,743 +1,322 @@
-# Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
-#
-# Licensed under the Apache License, Version 2.0 (the "License").
-# You may not use this file except in compliance with the License.
-# A copy of the License is located at:
-#
-#    http://aws.amazon.com/apache2.0/
-#
-# or in the "license" file accompanying this file. This file is
-# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS
-# OF ANY KIND, either express or implied. See the License for the
-# specific language governing permissions and limitations under the
-# License.
+#!/usr/bin/env python3
+
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
 """A repeatable benchmark tool for ion-python implementation.
 
 Usage:
-    ion_python_benchmark_cli.py write [--results-file <path>] [--api <api>]... [--c-extension <bool>] [--warmups <int>] [--iterations <int>] [--format <format>]... [--io-type <io_type>]... <input_file>
-    ion_python_benchmark_cli.py read [--results-file <path>] [--api <api>]... [--iterator <bool>]  [--c-extension <bool>] [--warmups <int>] [--iterations <int>] [--format <format>]... [--io-type <io_type>]... <input_file>
-    ion_python_benchmark_cli.py compare (--benchmark-result-previous <file_path>) (--benchmark-result-new <file_path>) <output_file>
-    ion_python_benchmark_cli.py (-h | --help)
+    ion_python_benchmark_cli.py (run|read|write|compare) [<args>]...
     ion_python_benchmark_cli.py (-v | --version)
+    ion_python_benchmark_cli.py (-h | --help)
 
 Command:
-    write       Benchmark writing the given input file to the given output format(s). In order to isolate
-    writing from reading, during the setup phase write instructions are generated from the input file
-    and stored in memory. For large inputs, this can consume a lot of resources and take a long time
-    to execute.
+    write       Benchmark writing the given input file to the given output format(s). In order to isolate writing from
+                reading, during the setup phase write instructions are generated from the input file and stored in
+                memory. For large inputs, this can consume a lot of resources and take a long time to execute.
+
+    read        First, re-write the given input file to the given output format(s) (if necessary), then benchmark
+                reading the resulting log files.
 
-    read        First, re-write the given input file to the given output format(s) (if necessary), then
-    benchmark reading the resulting log files.
+    run         Run benchmarks for a benchmark spec file.
```

Comment:
suggest `spec` what do you think?

Reply:
Sure. I was going for a verb, like `read`, `write`, and `compare`, but I wasn't really happy with `run` because it is too vague.